src/screens/welcome/LoginScreen.tsx

```diff
@@ -61,19 +85,46 @@ const LoginScreen = ({ navigation }: Props) => {
         errorText={password.error}
         secureTextEntry
       />
-
       <View style={styles.forgotPassword}>
         <TouchableOpacity
           onPress={() => navigation.navigate('ForgotPasswordScreen')}
         >
           <Text style={styles.label}>Forgot your password?</Text>
         </TouchableOpacity>
       </View>
-
-      <Button mode="contained" onPress={_onLoginPressed}>
-        Login
-      </Button>
-
+      <Mutation mutation={LOGIN_USER}>
+        {(login, { data, error }) => (
+          <View style={styles.submitContainer}>
+            {error && (
+              <Text style={styles.error}>
+                {error.graphQLErrors[0]
+                  ? error.graphQLErrors[0].message
+                  : 'Error'}
+              </Text>
+            )}
+            <Button
+              mode="contained"
+              onPress={() => {
+                if (validateInput()) {
+                  login({
+                    variables: {
+                      email: email.value,
+                      password: password.value,
+                    },
+                  }).then(response => {
+                    if (response.data) {
+                      loginThunk(response.data);
+                      //there is no need to navigate the user. The navigation container will change to signedIn from redux props
+                    }
```

Comment:
Maybe move this stuff to separate functions, it's a bit too much code to have inline